includes/model/class-openedx-woocommerce-plugin-enrollment.php

```diff
@@ -0,0 +1,785 @@
+     */
+    public function register_status() {
+        register_post_status(
+            'eor-success',
+            array(
+                'label'                     => __( 'Success', 'wp-edunext-marketing-site' ),
+                'public'                    => false,
+                'internal'                  => true,
+                'private'                   => true,
+                'exclude_from_search'       => false,
+                'show_in_admin_all_list'    => true,
+                'show_in_admin_status_list' => true,
+                'label_count'               => _n_noop( 'Success <span class="count">(%s)</span>', 'Success <span class="count">(%s)</span>', 'wp-edunext-marketing-site' ),
+            )
+        );
+        register_post_status(
+            'eor-pending',
+            array(
+                'label'                     => __( 'Pending', 'wp-edunext-marketing-site' ),
+                'public'                    => false,
+                'internal'                  => true,
+                'private'                   => true,
+                'exclude_from_search'       => false,
+                'show_in_admin_all_list'    => true,
+                'show_in_admin_status_list' => true,
+                'label_count'               => _n_noop( 'Pending <span class="count">(%s)</span>', 'Pending <span class="count">(%s)</span>', 'wp-edunext-marketing-site' ),
+            )
+        );
+        register_post_status(
+            'eor-error',
+            array(
+                'label'                     => __( 'Error', 'wp-edunext-marketing-site' ),
+                'public'                    => false,
+                'internal'                  => true,
+                'private'                   => true,
+                'exclude_from_search'       => false,
+                'show_in_admin_all_list'    => true,
+                'show_in_admin_status_list' => true,
+                'label_count'               => _n_noop( 'Error <span class="count">(%s)</span>', 'Error <span class="count">(%s)</span>', 'wp-edunext-marketing-site' ),
+            )
+        );
+    }
+
+    /**
+     * Wrapper for the WP function that prevents an infinite cycle of hook recursion
+     *
+     * @param array $post The post info in an array.
+     */
+    public function wp_update_post( $post ) {
+        $this->unregister_save_hook();
+
+        wp_update_post( $post );
+
+        $this->register_save_hook();
+    }
+
+    /**
+     * Save post metadata when a post is saved.
+     *
+     * @param int  $post_id The post ID.
+     * @param post $post The post object.
+     * @param bool $update Whether this is an existing post being updated or not.
+     */
+    public function save_action( $post_id, $post, $update ) {
+
+        if ( $this->post_type !== $post->post_type ) {
+            return;
+        }
+
+        $oerarr = array(
+            'oer_course_id'    => sanitize_text_field( $_POST['oer_course_id'] ),
```

Comment:
Back in the day I wrote the oer_<field> meta data fields with the oer prefix. I think it ultimately became confusing. OER was Openedx Enrollment Request. Please feel free to change this to the longer "enrollment" prefix or choose any other prefix that is less confusing that oer.

Reply:
Yes, totally agree. I am going to proceed with the replacement of "oer" to "enrollment"